### PR TITLE
Update Weights for Vesting Pallet

### PR DESCRIPTION
**Pull request by shawntabrizi**
This is a PR that updates the weights for the vesting pallet based on benchmarking data collected.

### CODE REVIEW

File: frame/vesting/src/lib.rs [hunk 1]
```diff
@@ -233,10 +234,13 @@ decl_module! {
 		/// Emits `VestingCreated`.
 		///
 		/// # <weight>
-		/// - Creates a new storage entry, but is protected by a minimum transfer
-		///	   amount needed to succeed.
+		/// - `O(1)`.
+		/// - DbWeight: 3 Reads, 3 Writes
```
Comment (marcio-diaz) [inline frame/vesting/src/lib.rs:238]: 3 or 4?

File: frame/vesting/src/lib.rs [hunk 2]
```diff
@@ -190,11 +190,12 @@ decl_module! {
...
-		/// - One balance-lock operation.
-		/// - One storage read (codec `O(1)`) and up to one removal.
-		/// - One event.
+		/// - DbWeight: 2 Reads, 2 Writes
+		///     - Reads: Vesting Storage, Balances Locks, [Sender Account]
+		///     - Writes: Vesting Storage, Balances Locks, [Sender Account]
+		/// - Benchmark: 147.5 µs (min square analysis)
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(MINIMUM_WEIGHT)]
+        #[weight = 150_000_000 + T::DbWeight::get().reads_writes(2, 2)]
```
Comment (thiolliere) [inline frame/vesting/src/lib.rs:198]:
```suggestion
		#[weight = 150_000_000 + T::DbWeight::get().reads_writes(2, 2)]
```

File: frame/vesting/src/lib.rs [hunk 3]
```diff
@@ -233,10 +234,13 @@ decl_module! {
...
-		/// - Creates a new storage entry, but is protected by a minimum transfer
-		///	   amount needed to succeed.
+		/// - `O(1)`.
+		/// - DbWeight: 3 Reads, 3 Writes
+		///     - Reads: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		///     - Writes: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		/// - Benchmark: 263 µs (min square analysis)
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(1_000_000_000)]
+        #[weight = 260_000_000 + T::DbWeight::get().reads_writes(4, 4)]
```
Comment (thiolliere) [inline frame/vesting/src/lib.rs:243]: we shouldn't count sender account I think
```suggestion
		#[weight = 260_000_000 + T::DbWeight::get().reads_writes(3, 3)]
```

File: frame/vesting/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply frame/vesting/src/lib.rs:198]: There are spaces!

File: frame/vesting/src/lib.rs [hunk 4]
```diff
@@ -211,12 +212,12 @@ decl_module! {
...
-		/// - One balance-lock operation.
-		/// - One storage read (codec `O(1)`) and up to one removal.
-		/// - One event.
+		/// - DbWeight: 3 Reads, 3 Writes
+		///     - Reads: Vesting Storage, Balances Locks, Target Account
+		///     - Writes: Vesting Storage, Balances Locks, Target Account
+		/// - Benchmark: 150.4 µs (min square analysis)
 		/// # </weight>
-		#[weight = MINIMUM_WEIGHT]
+        #[weight = 150_000_000 + T::DbWeight::get().reads_writes(3, 3)]
```
Comment (bkchr) [inline frame/vesting/src/lib.rs:220]: Spaces.

File: frame/vesting/src/lib.rs (hunk 3 above)
Comment (bkchr) [reply frame/vesting/src/lib.rs:243]: Spaces as well.

**Comment by shawntabrizi** [comment]
Okay this is ready for review. Should be good based on the benchmark data.

File: frame/vesting/src/lib.rs [hunk 5]
```diff
@@ -233,10 +236,14 @@ decl_module! {
...
-		///	   amount needed to succeed.
+		/// - `O(1)`.
+		/// - DbWeight: 3 Reads, 3 Writes
+		///     - Reads: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		///     - Writes: Vesting Storage, Balances Locks, Target Account, [Sender Account]
+		/// - Benchmark: 263 µs (min square analysis)
+		/// - Assuming less than 50 locks on any user, else we may want factor in number of locks.
 		/// # </weight>
-		#[weight = 1_000_000_000]
+		#[weight = 300_000_000 + T::DbWeight::get().reads_writes(3, 3)]
```
Comment (kianenigma) [inline frame/vesting/src/lib.rs:246]: do we have a rule of thumb for granularity or soundings? This, accurately, should have been around 260_000_000 or sth?

Comment (shawntabrizi) [reply frame/vesting/src/lib.rs:246]: Benchmarks are not that accurate to begin with, but yeah, rounding here is mostly just an opinion, as long as it does not extremely change the value.

Comment (shawntabrizi) [reply frame/vesting/src/lib.rs:246]: Benchmarks are not that accurate to begin with, but yeah, rounding here is mostly just an opinion, as long as it does not extremely change the value.

I am also taking into consideration the variability of locks. If this is a little bit larger, it is a little bit safer overall as more locks are introduced per user.

**Comment by gnunicorn** [comment]
bot merge

**Pull request merged**